addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,171 @@
+
+        lead_values = {
+            'contact_name': username or user_input_lines_lead_data["user_nickname"],
+            'description': user_input_lines_lead_data["description"],
+            'name': f"{lead_title_start} {_('live session results')}" if self.is_session_answer else f"{lead_title_start} {_('survey results')}",
+            'user_id': survey_responsible.id,
+        }
+
+        # Associate lead to the existing partner when known. Either because the person is connected
+        # or because they received the survey with a unique token (by email for example).
+        if self.partner_id:
+            lead_values['partner_id'] = self.partner_id.id
+        else:  # Save email field answer otherwise
+            lead_values['email_from'] = user_input_lines_lead_data["public_user_mail"]
+
+        return lead_values
+
+    def _prepare_lead_values_from_user_input_lines(self):
+        """ This method will prepare necessary lead values (in a dictionary), from the
+            content of user input lines, to create a lead: it will format the lead
+            description and get user's nickname and his email, if they're provided.
+            To write the description (notes in CRM) in HTML format, there are 5 cases:
+                - Suggested answers question: Question — Answer 1, Answer 3
+                    (==> expected html line: <li>Question — Answer 1, Answer 3</li>)
+                - Matrix question:  Question
+                                    Line label 1 — Answer 1
+                                    Line label 1 — Answer 3
+                                    Line label 4 — Answer 2
+                    (==> expected html line: <li>
+                                                Question
+                                                <br>&esmpLine label 1 — Answer 1
+                                                <br>&esmpLine label 1 — Answer 3
+                                                <br>&esmpLine label 4 — Answer 2
+                                             </li>)
+                - Long text question: Question
+                                      Text line 1
+                                      Text line 2
+                                      Text line 3
+                    (==> expected html line: <li>
+                                                Question
+                                                <br>&esmpText line 1
+                                                <br>&esmpText line 2
+                                                <br>&esmpText line 3
+                                             </li>)
+                - Other types: Question — Answer
+                    (==> expected html line: <li>Question — Answer</li>)
+                - Skipped question: Question — Skipped
+                    (==> expected html line: <li>Question — <i>Skipped</i></li>)
+
+            Note: "<br/>&emsp;" is used to get a line break with indentation.
```

Comment:
Nice :) I would only keep the html for each, as it is a bit redundant. It is code, so it can be technical ^^